x/interchainqueries/types/registered_query.go

```diff
@@ -0,0 +1,15 @@
+package types
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+)
+
+func (queryInfo *RegisteredQuery) GetOwnerAddress() (creator sdk.AccAddress, err error) {
+	creator, errCreator := sdk.AccAddressFromBech32(queryInfo.Owner)
+	return creator, sdkerrors.Wrapf(errCreator, ErrInvalidOwner.Error(), queryInfo.Owner)
```

Comment:
I don't think we need another error like `ErrInvalidOwner` cause we use `sdkerrors.ErrInvalidAddress` to handle errors with addresses, so I suggest:

```suggestion
	return creator, sdkerrors.Wrapf(sdkerrors.ErrInvalidAddress, "failed to decode owner address: %s", msg.Sender)
```